packages/editor/src/components/routes/link.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs, getQueryArgs, removeQueryArgs } from '@wordpress/url';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
```

Comment:
I don't think the "editor" package should depend on the "router". We should consider the "editor" package as a generic package to render `<Editor postType={ postType } postId={ postId } />` for instance we could have multiple on the same page. Based on that I think we need two optional settings to the "EditorProvider" to address the use-case of this PR:

 - onSelectPost( postType, postId ): a setting to call to navigate to a new postType/postId.
 - goBack (or something like that): to go back to the previous edited entity.

Worth noting that the document bar is not the only place where we'd want to use these settings, I can think of the "template panel" as another place to use onSelectPost for instance.